sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CastSuite.scala

```diff
@@ -961,7 +961,7 @@ abstract class AnsiCastSuiteBase extends CastSuiteBase {
   }
 
   test("ANSI mode: cast string to timestamp with parse error") {
-    val activeConf = conf
+    val activeConf = conf.clone()
```

Comment:
do we have to clone the conf here?

Reply:
agree, I thought that this was a tentative workaround.

Reply:
@dongjoon-hyun Can #31775 allow us to revert this?

Reply:
I think so.